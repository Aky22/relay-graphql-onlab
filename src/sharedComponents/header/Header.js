import React, { Component } from 'react';
import styled from 'styled-components';

import HeaderTitle from './components/HeaderTitle';

const Container = styled.div`
  clear: both;
`;

class Header extends Component {
  constructor(props) {
    super(props);

    this.currentRoute = window.location.pathname;
    this.excludedRoutes = [
      '/page-without-header-example',
    ];
  }

  
  render() {
    return (
      <Container>
        {this.excludedRoutes.includes(this.currentRoute) ? null : (
          <HeaderTitle>
            I'm the header
          </HeaderTitle>
        )}
      </Container>
    );
  }
}

export default Header;
