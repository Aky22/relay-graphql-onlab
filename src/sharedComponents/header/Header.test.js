import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Header from './Header';

Enzyme.configure({ adapter: new Adapter() });

describe('Header component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Header />
    );
    expect(wrapper).toMatchSnapshot();
  });
});