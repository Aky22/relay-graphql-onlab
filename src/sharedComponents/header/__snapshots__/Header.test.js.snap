// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<styled.div>
  <HeaderTitle>
    I'm the header
  </HeaderTitle>
</styled.div>
`;
